zygrader/data/model.py

```diff
@@ -372,17 +372,21 @@ def compile_and_run_code(self, use_gdb):
         curses.doupdate()
         return True
 
-    def pick_part(self, title="Choose a part"):
+    def pick_part(self, title="Choose a part", pick_all=False):
         window = ui.get_window()
         part_names = [self.get_part_identifier(x) for x in self.lab.parts]
 
         popup = ui.layers.ListLayer(title, popup=True)
+        if pick_all:
+            popup.add_row_text("Pick all latest")
         for part in part_names:
             popup.add_row_text(part)
         window.run_layer(popup)
         if popup.was_canceled():
             return None
 
+        if pick_all:
+            return popup.selected_index() - 1
         return popup.selected_index()
```

Comment:
This also needs a -1 right? So that would mean we can just `return popup.selected_index() - 1` regardless if `pick_all` is true

Reply:
I don't 100% know how the popups work so correct me if I'm wrong but it seems the index relevant is simply determined by the order added. As the "all latest" option is only added conditionally, when `pick_all` is false the -1 is not needed

Reply:
Derp. yep :)